core/src/test/java/com/google/cloud/sql/core/InternalConnectorRegistryTest.java

```diff
@@ -367,6 +368,107 @@ public void testGetApplicationNameFailsAfterInitialization() {
         () -> InternalConnectorRegistry.setApplicationName("sample-app"));
   }
 
+  @Test
+  public void registerConnection() throws IOException, InterruptedException {
+    FakeSslServer sslServer = new FakeSslServer();
```

Comment:
There's a lot of duplicate setup here. Could we consolidate it to make these tests easier to read?